lib/edusign/client.rb

```diff
@@ -5,13 +5,27 @@ class Client
     include HTTParty
     base_uri "https://ext.edusign.fr/v1"
 
+    class << self
+      def setup
+        yield self
+      end
+
+      def account_api_key=(key)
+        @@account_api_key = key
+      end
+    end
+
     ALREADY_LOCKED_ERROR_MESSAGE = "Course already locked".freeze
     STUDENT_ALREADY_ADDED_TO_COURSE_ERROR_MESSAGE = "Student already in the list"
     LOCKED_ERROR_MESSAGE = "course locked".freeze
 
+    class NoApiKeyError < StandardError; end
+
     class BadGatewayError < StandardError; end
 
-    def initialize(account_api_key:)
+    def initialize(account_api_key: @@account_api_key)
```

Comment:
Not a big fan of the `@@` variable, maybe you could add `activesupport` as a runtime dependency to the gem and use [ActiveSupport::Configurable](https://api.rubyonrails.org/classes/ActiveSupport/Configurable.html) ?

```ruby
require "active_support/configurable"

module Edusign
  class Client
    include ActiveSupport::Configurable

    def initialize(account_api_key:)
      @account_api_key = account_api_key || config.account_api_key

      raise NoApiKeyError, "Please provide an Edusing account API key" if @account_api_key.nil?
    end
  end
end
```

```ruby
# In an initializer in Rails app using that gem:

Edusign::Client.configure do |config|
  config.account_api_key = "THE KEY"
end
```

WDYT?

Reply:
Sounds good 👍 

Out of curiosity, why you don't like `@@` variables? (I was inspired by what [devise do](https://github.com/heartcombo/devise/blob/main/lib/devise.rb#L68))

Reply:
I'd rather encapsulate all configuration variables (I know there is one, for now) into a dedicated object than multiplying the class variables.